docs/ha_3scale.adoc

```diff
@@ -0,0 +1,216 @@
+
+=== Scaling number of replicas
+
+The number of replicas of each component needs to be at least 2 to support pod failures. For installations with nodes running in different availability zones, the minimum replica count should match the number of availability zones available. This configuration used in combination with node-based pod anti-affinity rules, described in the next section, will ensure that at least one pod pod will run in each availability zone. 3scale components that support more than one replicas are:
+
+- apicast-production
+- apicast-staging
+- backend-listener
+- backend-worker
+- system-app
+- system-sidekiq
+- zync
+- zync-que
+
+Depending on your cluster size, throughput and latency requirements you might want to deploy more replicas, at least for the components that are called on each request to the APIs configured in 3scale, that is, `apicast-production`, `backend-listener`, and `backend-worker`.
+
+Replicas can be configured using the `replicas` attribute in the `APIManager` spec.
+
+=== Pod Affinity
+
+Enabling pod affinities in the operator for every component, ensures that pod replicas from each deployment are distributed across different nodes of the cluster, and evenly balanced across different availability zones. You can read more about affinities in the link:https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity[Kubernetes docs].
+
+To enable affinities you need to follow link:https://github.com/3scale/3scale-operator/blob/master/doc/operator-user-guide.md#setting-custom-affinity-and-tolerations[these examples].
+
+The following example to be added for example to `apicastProductionSpec` (but can be added to any non-database deployment), adds a *soft podAntiAffinity* configuration using `preferredDuringSchedulingIgnoredDuringExecution` (so scheduler will try to run this set of `apicast-production` pods in different hosts from different availability zones, but if it is not possible, then allow to run elsewhere):
+```yaml
+      affinity:
+        podAntiAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+            - weight: 100
+              podAffinityTerm:
+                labelSelector:
+                  matchLabels:
+                    deploymentConfig: apicast-production
+                topologyKey: kubernetes.io/hostname
+            - weight: 99
+              podAffinityTerm:
+                labelSelector:
+                  matchLabels:
+                    deploymentConfig: apicast-production
+                topologyKey: topology.kubernetes.io/zone
+```
+In the following example, unlike previous one, it sets a *hard podAntiAffinity* configuration using `requiredDuringSchedulingIgnoredDuringExecution` (so conditions must be met for a pod to be scheduled onto a node, otherwise pod won’t be scheduled, which can be risky on certain situations like being on a cluster with low free resources, and new pods cannot be scheduled):
```

Comment:
```suggestion
In the following example, unlike in the previous one, a *hard podAntiAffinity* configuration is set using `requiredDuringSchedulingIgnoredDuringExecution` (so conditions must be met for a pod to be scheduled onto a node, otherwise pod won’t be scheduled, which can be risky on certain situations like being on a cluster with low free resources, and new pods cannot be scheduled):
```

Reply:
fixed